src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -69,6 +69,7 @@
 #include "runtime/fieldDescriptor.hpp"
 #include "runtime/fieldDescriptor.inline.hpp"
 #include "runtime/frame.inline.hpp"
+#include "runtime/globals.hpp"
```

Comment:
Is this extra include needed?

Reply:
Not with the current state of the codebase. Currently we get  `"utilities/globalDefinitions.hpp"` and `"runtime/globals.hpp"` via the following include chains:
```
#include "classfile/javaClasses.inline.hpp"
  #include "classfile/javaClasses.hpp"
    #include "classfile/vmClasses.hpp"
      #include "classfile/vmClassID.hpp"
        #include "utilities/enumIterator.hpp"
          #include "metaprogramming/primitiveConversions.hpp"
            #include "utilities/globalDefinitions.hpp"
    #include "runtime/handles.hpp"
      #include "memory/arena.hpp"
        #include "runtime/globals.hpp"
```

But in general, especially for `.cpp` files, I like to include directly what is used (except for system headers like `cstdint` and `cstddef` for which I include `utilities/globalDefinitions.hpp`). All this is to have removals elsewhere not break the includes (it can obviously still break things due to updated include order if we do nasty things with the preprocessor etc.)

However I know that there is sometimes pushback on both `globals.hpp` and `globalDefinitions.hpp` because for almost any non-trivial compilation unit (that is not utility code) they will get included from some of its leaf dependencies. 

I'll drop these includes in this PR. But that is my rational for including them